front/lib/tracking/customerio/server.ts

```diff
@@ -0,0 +1,175 @@
+import type {
+  LightWorkspaceType,
+  MembershipRoleType,
+  UserType,
+  UserTypeWithWorkspaces,
+} from "@dust-tt/types";
+import * as _ from "lodash";
+
+import { subscriptionForWorkspace } from "@app/lib/auth";
+import { Workspace } from "@app/lib/models/workspace";
+import { MembershipResource } from "@app/lib/resources/membership_resource";
+import { renderLightWorkspaceType } from "@app/lib/workspace";
+
+const {
+  CUSTOMERIO_ENABLED,
+  CUSTOMERIO_SITE_ID = "",
+  CUSTOMERIO_API_KEY = "",
+} = process.env;
+
+const CUSTOMERIO_HOST = "https://track-eu.customer.io/api/v2";
+
+function isCustomerioEnabled() {
+  return CUSTOMERIO_ENABLED === "true";
+}
+
+export class CustomerioServerSideTracking {
+  static trackSignup({ user }: { user: UserType }) {
```

Comment:
Here you are delcaring the type as a `UserType` but in reality we are using a `UserTypeWithWorkspaces`, so you don't 
have to fetch the workspaces again.
As you know we are movign away from using models outside of the "resource layer", which does not exist yet for workspaces, 
but I had already made a pass on removing direct access to the storage layer in tracking, so lets make sure to not re-introduce it :pray:

Reply:
Actually not the same thing. Here I need to fetch all memberships, because user.workspaces does not include ended memberships and I need those.